### ISSUE TITLE
Core package does not transpile with strict tsconfig

### ISSUE THREAD

**Issue by Danchieke** [amcharts/amcharts5#771]
**Bug description**

An angular project does not compile with amCharts unless `strictNullChecks` and `strictFunctionTypes` are both disabled. I assume this would also be true of other non-angular projects with strict typing.

Error message:
```
Error: node_modules/@amcharts/amcharts5/.internal/charts/hierarchy/ForceDirected.d.ts:166:15 - error TS2416: Property '_updateValues' in type 'ForceDirected' is not assignable to the same property in base type 'LinkedHierarchy'.
  Type '(d3HierarchyNode: HierarchyNode<IForceDirectedDataObject>) => void' is not assignable to type '(d3HierarchyNode: HierarchyNode<IHierarchyDataObject>) => void'.
    Types of parameters 'd3HierarchyNode' and 'd3HierarchyNode' are incompatible.
      Type 'HierarchyNode<IHierarchyDataObject>' is not assignable to type 'HierarchyNode<IForceDirectedDataObject>'.
        Types of property 'data' are incompatible.
          Type 'IHierarchyDataObject' is not assignable to type 'IForceDirectedDataObject'.

166     protected _updateValues(d3HierarchyNode: d3Hierarchy.HierarchyNode<IForceDirectedDataObject>): void;
                  ~~~~~~~~~~~~~
Error: node_modules/@amcharts/amcharts5/.internal/charts/hierarchy/LinkedHierarchy.d.ts:19:18 - error TS2430: Interface 'ILinkedHierarchyDataItem' incorrectly extends interface 'IHierarchyDataItem'.

19 export interface ILinkedHierarchyDataItem extends IHierarchyDataItem {
                    ~~~~~~~~~~~~~~~~~~~~~~~~
```
The error continues much further down the inheritance structure.

**Reproduction steps**
- Create a new Angular 15 project (by default the tsconfig will have `strict: true`
- Install amCharts (`npm i @amcharts/amcharts5`)
- Import the amCharts core library into the app.component.ts
- Build the project

**Environment**
OS: Windows 10
AmCharts: tested 5.2.15 and 5.2.50
Angular: 15.1.0
Node: 18.13.0 (lts)

**Comment by lucabonvinigug** [comment]
Same problem since yesterday.

**Environment**
OS: tested on MacOs and dockerized Ubuntu Server
AmCharts: tested from 5.2.10 to 5.2.50
Angular: tested 12.2.11 and 14.2.10
Node: tested 14.17.3, 16.15.1 and latest (alpine)

**Comment by Pauan** [comment]
Strange, we run all of our compilations with strict mode, and we also test with Angular. We'll look into it.

**Comment by renkei** [comment]
Strange, we have the option `strict: true` for a long time, our

Angular 14, TS 4.7, amcharts 5.2

project compiled fine with this option. But after switching to

Angular 15, TS 4.9, amcharts 5.3

we also get a lot of compiler errors as described above. Downgrade of amcharts to version 5.2 does not help, downgrade to TS 4.8 does not help, downgrade to TS 4.7 is not possible, because of missing support by Angular 15 (requires TS >= 4.8.2).

**Comment by renkei** [comment]
I've found the cause of the issue: It's `@types/d3-hierarchy@3.1.2`, if you downgrade to `@types/d3-hierarchy@3.1.1`, all compiler errors are gone.

**Comment by Pauan** [comment]
@renkei That's really helpful, thanks! I can reproduce this error, and we're looking into possible solutions or workarounds.

**Comment by martynasma** [comment]
Fixed in [5.3.1](https://github.com/amcharts/amcharts5/releases/tag/5.3.1).

## [5.3.1] - 2023-01-31

### Added
- `mainDataItems` getter was added to `XYSeries`. Returns ungrouped data items array.

### Changed
- `crisp` is now set to `true` by default in `Tick` elements.

### Fixed
- If all values in a grouped period were null, series showed 0 instead of omiting data point when group value was set to "sum".
- Tooltip of a series with `maxTooltipDistance: -1` could be hidden when it should have been shown.
- Setting custom `stockSeries` on an indicator was not working (Chart was force-setting its own `stockSeries` on indicators).
- Fixing type error with `@types/d3-hierarchy`.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
